File: LinkedList/Java/Problem4.java
/*
Delete from the last of Singly Linked List.
*/

public static Node deleteEnd(Node head) {
        if((head == null) || (head.next == null)) { return null; }
        Node curr = head;
        while(curr.next.next != null)
            curr = curr.next;
        curr.next = null;
        return  head;
    }

File: LinkedList/Cpp/Problem4.cpp
/*
Delete from the end of the singly Linked List.
*/

struct Node
{
  int data;
  Node *next;
  Node(int x)
  {
    data = x;
    next = NULL;
  }
};

Node *deleteEnd(Node *head)
{
  if (head == NULL)
    return NULL;
  if (head->next == NULL)
  {
    delete head;
    return NULL;
  }
  Node *curr = head;
  while (curr->next->next != NULL)
    curr = curr->next;
  delete (curr->next);
  curr->next = NULL;
  return head;
}
